include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp

```diff
@@ -150,6 +150,33 @@ static std::optional<unsigned> getBitWidth(Type a) {
   return std::nullopt;
 }
 
+static void processTwoStridesLastDim(RankedTensorType type, ConversionPatternRewriter &rewriter,
+    std::function<void(SmallVector<OpFoldResult> &/* offsets */,
+        SmallVector<OpFoldResult>&/* sizes */,
+        SmallVector<OpFoldResult>&/* strides */,
+        int /* index */)> op) {
+  int64_t rank = type.getRank();
+  auto shape = type.getShape();
+
+  SmallVector<OpFoldResult> offsets, sizes, strides;
+  SmallVector<int64_t> stridesInt(rank, 1);
+
+  for (size_t j = 0; j < shape.size(); ++j) {
+    offsets.push_back(rewriter.getIndexAttr(0));
+    sizes.push_back(rewriter.getIndexAttr(shape[j]));
+    strides.push_back(rewriter.getIndexAttr(stridesInt[j]));
+  }
+
+  for (int i = 0; i < 2; ++i) {
+    offsets.pop_back();
+    sizes.pop_back();
+
+    offsets.push_back(rewriter.getIndexAttr(i));
+    sizes.push_back(rewriter.getIndexAttr(1));
+    op(offsets, sizes, strides, i);
+  }
+}
+
```

Comment:
I still have some reservations about this function. I understand that it's being used to share code between split and join, but the purpose seems very limited and doesn't offer reuse elsewhere.

At least for me, it makes reading the implementation for split & join much harder since I have to go back and forth between the call site and this implementation. I think the code that we save by having this function is not that much: only the last loop seems non-trivial. The first loop can actually just be simplified by initializing the vector using constructor (except for the sizes). For instance:

```cpp
SmallVector<OpFoldResult> offsets(shape.size(), rewriter.getIndexAttr(0));
SmallVector<OpFoldResult> strides(shape.size(), rewriter.getIndexAttr(1));
SmallVector<OpFoldResult> sizes = llvm::map(...)
```

I think with the above reasons, it might make the code more readable by placing the code directly in the Converter. Let me know what you think.

Reply:
no problem, updated and substituted, thank you!